src/components/Filters/tests/Filters.test.tsx

```diff
@@ -464,6 +465,40 @@ describe('<Filters />', () => {
       );
       expect(rightActionButton.text()).toBe('More filters (2)');
     });
+
+    it('calls clear all callback and shifts focus when clear all filters is clicked to prevent visual loss of focus', () => {
+      const focusSpy = jest.spyOn(focusUtils, 'focusFirstFocusableNode');
+      const spy = jest.fn();
+      const appliedFilters = [{key: 'filterOne', label: 'foo', onRemove: spy}];
+
+      const resourceFilters = mountWithAppProvider(
+        <Filters {...mockProps} appliedFilters={appliedFilters} />,
+      );
+      resourceFilters.setProps({
+        onClearAll: () => {
+          resourceFilters.setProps({
+            appliedFilters: [],
+          });
+        },
+      });
+
+      trigger(findByTestID(resourceFilters, 'SheetToggleButton'), 'onClick');
+
+      let clearAllButton = resourceFilters
+        .findWhere((node) => node.prop('children') === 'Clear all filters')
+        .first();
+
+      expect(clearAllButton.prop('disabled')).toBeFalsy();
+
+      trigger(clearAllButton, 'onClick');
+
+      clearAllButton = resourceFilters
+        .findWhere((node) => node.prop('children') === 'Clear all filters')
+        .first();
+
+      expect(clearAllButton.prop('disabled')).toBeTruthy();
+      expect(focusSpy).toHaveBeenCalled();
```

Comment:
```suggestion
      expect(focusSpy).toHaveBeenCalled();
      focusSpy.mockRestore();
```

I think we'll want to clear the mock

Reply:
Oops, good catch!